# Spike Phage DMS Repo

This repository contains all necessary code for analysis described in <High resolution profiling of pathways of escape for SARS-CoV-2 spike-binding antibodies by Garrett et al>
The repository contains the code for generating the sequences used to make the Spike Phage-DMS library, a [Nextflow](https://www.nextflow.io/docs/latest/getstarted.html) pipeline,
dubbed [phip-flow](https://github.com/matsengrp/phip-flow) to align illumina sequencing reads, along with all analysis and plotting code.

### Abstract

TODO

### Spike Phage-DMS library design

All code used to optimize sequences and generate oligonucleotides to create the Spike Phage-DMS library can be found within the `library-design` directory. See README file within that directory for a description of the files needed and scripts run.

### phip-flow analysis environment

The library design script and analysis notebooks contain all code needed to run analysis 
for the manuscript using custom code from 
[phippery](https://github.com/matsengrp/phippery) along with a few other popular python packages.

We suggest using [conda](https://www.anaconda.com/) to create the environment like so:
```
conda env create -f environment.yml && conda activate pan-cov-manuscript
mkdir -p _ignore && cd _ignore
git clone git@github.com:matsengrp/phippery.git
cd phippery && python setup.py install --user && cd ../../
```

### Final analysis and plotting of data

All code used to wrangle and plot data to generate figures for publication can be found within the `analysis-and-plotting` directory, and requires the `Curated_Samples_Phage_DMS_Analysis.csv` output file generated by the phip-flow pipeline.


