---
title: "Spike Phage-DMS analysis and plots for manuscript"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Description of code
This code was written by Meghan Garrett to generate plots for the manuscript "High resolution profiling of pathways of escape for SARS-CoV-2 spike-binding antibodies" by Garrett et al. The required files for this analysis are:
* A curated .csv file (Curated_Samples_Phage_DMS_Analysis.csv) containing the enrichment values and scaled differential selection values from Phage-DMS experiments with either COVID patient plasma or healthy serum. 
* A .csv file (sample_correlation.csv) containing the correlations between biological replicates, generated after the alignment step but before averaging replicates. 
* A .tsv file from LANL [here](https://cov.lanl.gov/components/sequence/COV/int_sites_tbls.comp?t=2&l=) containing the mutational entropy values for all sites on Spike, based on GISAID data. The version used for this analysis was downloaded on 11/9/20.


## Import relevant packages
```{r}
library(tidyverse)
library(ggpubr)
library(corrr)
library(cowplot)
library(scales)
library(rstatix)
library(coin)
library(tidyselect)
library(ggrepel)
```

## Import and reshape data
```{r}
master <-read_csv("Curated_Samples_Phage_DMS_Analysis.csv")
```


Then, need to make sure the "Loc" is actually the correct locus number. Need to add 685 onto the S2 numbers in order to make them correct (because they start the number at 1, which isn't correct)
```{r}
master <- master %>% mutate(Correct_Loc = ifelse(Protein == "S2", Loc + 685, Loc))
```

Also need to make a column that has the sample timepoint and participant ID together
```{r}
#First make a column with just the sample day
master <- master %>% mutate(sample_day = case_when(patient_status == "conv outpatient 30d" ~ "30d", patient_status == "conv outpatient 60d" ~ "60d"))

#then concatenate them together
master$participant_timepoint = paste(master$participant_ID, master$sample_day, sep=", ")
```

Need to join the table with correlation between biological replicates to the master table
```{r}
corr_table <- read_csv("sample_correlation.csv")

master <- left_join(master, corr_table)
```


Also need to reorder the amino acids in the desired order
```{r}
master$aa_sub <- factor(master$aa_sub, levels = c("G", "P", "A", "I", "L", "V", "F", "W", "M", "Y", "C", "S", "T", "N", "Q", "H", "K", "R", "D", "E"))
```

Need to have each participant treated as a character and not a continuous number, then need to specify the correct order to number participants
```{r}
master$participant_ID <- as.factor(master$participant_ID)

master <- master %>% mutate(participant_ID = fct_relevel(participant_ID, "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18"))
```



Define a color palette for each participant. These were taken from "http://medialab.github.io/iwanthue/"
```{r}
Color_ID <- c("1" = "#e0366f","2" = "#5eba48", "3" = "#7c5cd1","4" = "#adb73d", "5" = "#6ca2da", "6" = "#9c5f2c", "7" = "#ca4583", "8" = "#5abe80", "9" = "#c781e2", "10" = "#d89a34", "11" = "#d05629", "12" = "#4dbfb7", "13" = "#e58f72", "14" = "#378053", "15" = "#8c4f9a", "16" = "#cb4aaf", "17" = "#6c943e", "18" = "#da85b6")

```


## Now, generate plots

**Plotting the range of correlation between biological replicate experiments**

```{r}
plot <- corr_table %>% ggplot(aes(x = patient_status, y=replicate_correlation, color = as.factor(participant_ID), shape = patient_status)) +
  geom_hline(yintercept=0.5, linetype="dashed") +
  geom_jitter(size=3, width=0.1) +
  theme_classic(base_size = 18) +
  expand_limits(y=c(0,1)) +
  xlab("") + 
  ylab("Pearson correlation (R) between biological replicates of a sample") +
  labs(shape = "Sample type", color = "Patient ID") +
  scale_shape(labels=c("30 day p.s.o.","60 day p.s.o.")) +
  scale_x_discrete(labels = c("30 day p.s.o.", "60 day p.s.o.")) +
  scale_color_manual(values = Color_ID) 

dir.create("correlation_matrix") # Folder to save plots into
dir <- "correlation_matrix/"

plot_name <- paste("correlation_matrix/biological_replicates_correlation.jpg")
ggsave(plot_name, plot, width = 8, height = 8)

print(plot)
```

**PLOTTING CORRELATION BETWEEN PAIRED SAMPLES - enrichment**

Get correlation matrix between paired samples vs unpaired samples. This plots a dotplot matrix with the correlations between all combinations.
```{r}

#make table wider with a unique column for each participant sample
master_corr <- master %>% 
  select(peptide_id, participant_timepoint, enrichment) %>% 
  pivot_wider(names_from = participant_timepoint, values_from = enrichment) %>%
  select(-peptide_id) %>%  
  select(str_sort(peek_vars())) 

corr_matrix <- master_corr %>% correlate() %>% shave() %>% print()

plot <- corr_matrix %>% rplot()

dir <- "correlation_matrix/"

plot_name <- paste("correlation_matrix/enrichment_paired_correlation_matrix.jpg")
ggsave(plot_name, plot, width = 10, height = 10)

```

Now, plot correlations between paired and unpaired samples in two columns.
```{r}
#Need to label each comparison as either between a paired or unparied sample. This searches to see if the first few letters in each participant column match each other, and creates a new column describing them as either paired (the correlation between two timepoints of the same participant) or unpaired (the correlation between random samples)
corr_table <- corr_matrix %>% 
  stretch() %>% 
  mutate(pair = if_else(substring(x, 1,3) == substring(y, 1, 3), "paired", "unpaired"))

# Remove NA columns (columns where two of the same samples are compared to each other)
corr_table <- corr_table %>% filter(r != "NA") %>% print()

#Now plot two columns with data points comparing paired and unpaired data
dp <- corr_table %>% ggplot(aes(x=pair, y=r)) +
  geom_boxplot(width=.6) +
  geom_jitter(position=position_jitter(0.2))+
  theme_classic(base_size = 18)+
  theme(legend.position = "none")  +
  xlab("") +
  ylab("Pearson Correlation (R)") + 
  stat_compare_means(label.x=1.4, label.y=1)

dir <- "correlation_matrix/"

plot_name <- paste("correlation_matrix/enrichment_paired_correlation.jpg")
ggsave(plot_name, dp, width = 6, height = 8)
```

Get the median correlation values from each plot
```{r}
corr_table %>% group_by(pair) %>% summarise(median = median(r)) %>% print()
```

Now, plot the relationship between paired correlation values and the sample's correlation btwn biological replicates
```{r}
replicate_corr_table <- master %>% 
  select(participant_timepoint, replicate_correlation, patient_status, participant_ID, sample_day) %>%
  unique() 
```
Rearrange the table to have each replicate with its paired correlation coefficient
```{r}
table_x <- corr_table %>% filter(pair == "paired") %>% select(-y, participant_timepoint = x)

table_y <- corr_table %>% filter(pair == "paired") %>% select(-x, participant_timepoint = y)

table <- bind_rows(table_x, table_y) %>% left_join(replicate_corr_table)


```
Now, plot
```{r}
plot <- table %>% ggplot(aes(x=r, y=replicate_correlation)) +
  geom_point(aes(shape = sample_day, color=participant_ID), size=3) +
  scale_color_manual(values = Color_ID) + 
  geom_smooth(method='lm', color="black", se=FALSE, size=1) +
  theme_classic(base_size = 18) +
  stat_cor() + 
  xlab("Correlation (R) between paired samples") + 
  ylab("Correlation (R) between biological\nreplicates of a sample") +
  labs(shape = "Sample timepoint", color = "Patient ID") +
  scale_shape(labels=c("30 day p.s.o.","60 day p.s.o."))
  

dir <- "correlation_matrix/"

plot_name <- paste("correlation_matrix/paired_correlation_vs_replicate_correlation_enrichment.jpg")
ggsave(plot_name, plot, width = 8, height = 6)

```



**Plot average enrichment**
Plot WT enrichment values as overlayed line plots and facet by patient status
```{r}
dir.create("enrichment_plots")

ID.labs <- c("30 days p.s.o", "60 days p.s.o.")
names(ID.labs) <- c("conv outpatient 30d", "conv outpatient 60d")

plota <- master %>% 
  filter(Protein != "S1D614G") %>% 
  filter(patient_status == "conv outpatient 30d") %>%
  filter(is_wt == TRUE) %>%
  ggplot( aes(x=Correct_Loc, y=enrichment, color=factor(participant_ID))) + 
    geom_line(alpha=0.8) + 
    theme_classic(base_size = 24) +
    theme(legend.position = "none", plot.margin = unit(c(0, 0, 0, 0), "cm"), strip.background = element_blank(), strip.text.x = element_text(face="bold", size = 24), panel.border = element_rect(colour = "black", fill=NA, size=1)) +
    facet_wrap(~patient_status, ncol=1, labeller = labeller(patient_status = ID.labs)) +
  xlab("") +
  ylab("")+
  ylim(0,80) +
  scale_color_manual(values = Color_ID) +
  annotate("rect", xmin = 245, xmax = 285, ymin = 0, ymax =80,
           alpha = .2,fill = "grey") +
  annotate("rect", xmin = 470, xmax = 510, ymin = 0, ymax = 80,
           alpha = .2,fill = "grey") +
  annotate("rect", xmin = 540, xmax = 580, ymin = 0, ymax = 80,
           alpha = .2,fill = "grey") +
  annotate("rect", xmin = 615, xmax = 650, ymin = 0, ymax = 80,
           alpha = .2,fill = "grey") +
  annotate("rect", xmin = 800, xmax = 840, ymin = 0, ymax = 80,
           alpha = .2,fill = "grey") + 
  annotate("rect", xmin = 1135, xmax = 1195, ymin = 0, ymax = 80,
           alpha = .2,fill = "grey") +
  scale_x_continuous(breaks=c(1,100,200,300,400,500,600,700,800,900,1000,1100, 1200), expand = c(0, 0)) + 
  guides(color = guide_legend(override.aes = list(size = 2)))

plotb <- master %>% 
  filter(Protein != "S1D614G") %>% 
  filter(patient_status == "conv outpatient 60d") %>%
  filter(is_wt == TRUE) %>%
  ggplot( aes(x=Correct_Loc, y=enrichment, color=factor(participant_ID))) + 
    geom_line(alpha=0.8) + 
    theme_classic(base_size = 24) +
    theme(legend.position = "none", plot.margin = unit(c(0, 0, 0, 0), "cm"), strip.background = element_blank(), strip.text.x = element_text(face="bold", size = 24), panel.border = element_rect(colour = "black", fill=NA, size=1)) +
    facet_wrap(~patient_status, ncol=1, labeller = labeller(patient_status = ID.labs)) +
  xlab("") +
  ylab("") +
  ylim(0,80)+
  scale_color_manual(values = Color_ID)+
  annotate("rect", xmin = 245, xmax = 285, ymin = 0, ymax =80,
           alpha = .2,fill = "grey") +
  annotate("rect", xmin = 470, xmax = 510, ymin = 0, ymax = 80,
           alpha = .2,fill = "grey") +
  annotate("rect", xmin = 540, xmax = 580, ymin = 0, ymax = 80,
           alpha = .2,fill = "grey") +
  annotate("rect", xmin = 615, xmax = 650, ymin = 0, ymax = 80,
           alpha = .2,fill = "grey") +
  annotate("rect", xmin = 800, xmax = 840, ymin = 0, ymax = 80,
           alpha = .2,fill = "grey") + 
  annotate("rect", xmin = 1135, xmax = 1195, ymin = 0, ymax = 80,
           alpha = .2,fill = "grey") +
  scale_x_continuous(breaks=c(1,100,200,300,400,500,600,700,800,900,1000,1100, 1200), expand = c(0, 0)) + 
  guides(color = guide_legend(override.aes = list(size = 2)))

#first combine the COVID patient plots
plot <- plot_grid(plota, plotb, align = "v", nrow = 2)

#Annotate the plot with x axis and y axis labels
plot <- annotate_figure(plot, bottom = text_grob("SARS-CoV-2 S Protein Position", size = 24), left = text_grob("Enrichment of WT Peptides", size = 24, rot = 90))

#Get legend
legend <- get_legend(plota + theme(legend.position="right") + labs(color = "Patient ID"))

#Add legend to plot
plot <- plot_grid(plot, legend, rel_widths = c(3, .4))


dir <- "enrichment_plots/"

plot_name <- paste(dir, "enrichment_patient_groups_line.jpg", sep = "")

ggsave(plot_name, plot, width = 24, height = 10)

```


Now plot the entire enrichment data as a heatmap

```{r}

ID.labs <- c("30 days p.s.o.", "60 days p.s.o.")
names(ID.labs) <- c("conv outpatient 30d", "conv outpatient 60d")

plota <- master %>% 
  filter(Protein != "S1D614G") %>% 
  filter(patient_status == "conv outpatient 30d") %>%
  filter(is_wt == TRUE) %>%
  ggplot( aes(x=Correct_Loc, y=fct_rev(participant_ID), fill=enrichment)) + 
    geom_tile() +
    facet_wrap(~patient_status, ncol=1, labeller = labeller(patient_status = ID.labs)) +
    scale_fill_gradient2(low="red", mid="white", high="darkblue", limits=c(0,120), na.value ="white") +
  theme_classic(base_size = 14) +
  theme(legend.position = "none", plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm"), axis.title.x=element_blank(), strip.background = element_blank(), strip.text.x = element_text(face="bold", size = 14), panel.border = element_rect(colour = "black", fill=NA, size=1)) +
    xlab("") +
    ylab("")+
  labs(fill = "Enrichment of \n WT peptides\n") +
  scale_x_continuous(breaks=c(1,100,200,300,400,500,600,700,800,900,1000,1100, 1200), expand = c(0, 0)) 

plotb <- master %>% 
  filter(Protein != "S1D614G") %>% 
  filter(patient_status == "conv outpatient 60d") %>%
  filter(is_wt == TRUE) %>%
    ggplot( aes(x=Correct_Loc, y=fct_rev(participant_ID), fill=enrichment)) + 
    geom_tile() +
    facet_wrap(~patient_status, ncol=1, labeller = labeller(patient_status = ID.labs)) +
    scale_fill_gradient2(low="red", mid="white", high="darkblue", limits=c(0,120), na.value ="white") +
  theme_classic(base_size = 14) +
  theme(legend.position = "none", plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm"), axis.title.x=element_blank(), strip.background = element_blank(), strip.text.x = element_text(face="bold", size = 14), panel.border = element_rect(colour = "black", fill=NA, size=1)) +
    xlab("") +
    ylab("")+
  scale_x_continuous(breaks=c(1,100,200,300,400,500,600,700,800,900,1000,1100, 1200), expand = c(0, 0)) 

#first combine the COVID patient plots
plot <- plot_grid(plota, plotb, align = "v", nrow = 2)

#Make legend, then add  
legend <- get_legend(plota + theme(legend.position="right"))

#Annotate the plot with x axis and y axis labels
plot <- annotate_figure(plot, bottom = text_grob("          SARS-CoV-2 S Protein Position", size = 14), left = text_grob("Patient ID", size = 14, rot = 90))

#Add legend to plot
plot <- plot_grid(plot, legend, rel_widths = c(3, .4))


dir <- "enrichment_plots/"

plot_name <- paste(dir, "enrichment_patient_groups_heatmap.jpg", sep = "")

ggsave(plot_name, plot, width = 20, height = 10)


```


**PLOTTING SCALED DIFF SELECTION HEATMAPS**

Plot the scaled diff selection with a heatmap, and make a facet-ed heatmap with only samples of interest.

Fusion peptide, selected patient samples, scaled diff sel
```{r}
dir.create("scaled_diff_select_plots")

ID.labs <- c("30 days p.s.o.", "60 days p.s.o.")
names(ID.labs) <- c("conv outpatient 30d", "conv outpatient 60d")

 plot1 <- master %>% 
    filter(Protein != "S1D614G") %>%
    filter(participant_ID %in% c("1", "3","8", "12")) %>%
    filter(Correct_Loc %in% 800:840) %>%
    ggplot( aes(x=Correct_Loc, y=aa_sub, fill=scaled_diff_sel, width=.90, height=.90)) + 
  geom_tile() +
  geom_point(aes(size=is_wt), stroke = 0) +
  scale_size_manual(values=c(0, 2)) +
  scale_fill_gradientn(colours = c("red","white","darkblue"), values = rescale(c(-360,0,100)), guide = "colorbar", limits=c(-360,100), na.value ="lightgrey") + 
  labs(x="SARS-CoV-2 S Protein Position", y="AA Substitution", fill = "Scaled differential \nselection value\n") +
  theme_bw(base_size=16) +
  facet_grid(patient_status ~ participant_ID, labeller = labeller(patient_status = ID.labs)) +
  theme(plot.margin = unit(c(-0.5, 0, 0, 0), "cm"), strip.background = element_blank(), strip.text.x = element_blank(), strip.text.y = element_text(face = "bold", size = 16), legend.title.align=0.5, panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
    rotate_x_text(angle=45) +
   guides(size = FALSE) + 
   scale_x_continuous(breaks = seq(800, 840, by = 5), expand = c(0.02,0.02)) 

dir <- "scaled_diff_select_plots/"
 
plot_name <- paste(dir, "FP_region_patients_scaleddiffsel.jpg", sep = "")
  
ggsave(plot_name, plot1, width = 20, height = 8)
```
Now make an enrichment plot for the selected patient samples within the FP region
```{r}
plot2 <- master %>% 
  filter(Protein != "S1D614G") %>% 
  filter(participant_ID %in% c("1", "3","8", "12")) %>%
  filter(is_wt == TRUE) %>%
  filter(Correct_Loc %in% 800:840) %>%
  ggplot( aes(x=Correct_Loc, y=enrichment, color=participant_ID, linetype=sample_day)) + 
  geom_line(alpha=0.8) + 
  theme_bw(base_size = 16) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  facet_wrap(~participant_ID, nrow=1) +
  ylab("Enrichment of WT peptides") +
  labs(linetype = "Sample timepoint") +
  ylim(-8,80) +
  scale_color_manual(values = Color_ID)  +
  guides(color = FALSE) +
  theme(plot.margin = unit(c(0, 0, 0, 0), "cm"), strip.background = element_blank(), strip.text.x = element_text(face="bold", size = 20), legend.title.align=0.5, panel.border = element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.title.x=element_blank()) +
   scale_linetype_discrete(labels=c("30 days p.s.o.", "60 days p.s.o."))+ 
   scale_x_continuous(breaks = seq(800, 840, by = 5), expand = c(0.02,0.02))+
   coord_cartesian(clip="off") +
   annotate("rect", xmin=815.6, xmax=833, ymin=-8, ymax=-2, alpha=0.4)+
  annotate("text", x=824.5, y=-5, label="FUSION PEPTIDE", fontface="bold")

dir <- "scaled_diff_select_plots/"

plot_name <- paste(dir, "FP_region_patients_enrichment.jpg", sep = "")

ggsave(plot_name, plot2, width = 20, height = 5)

```

Now, combine the two plots into one
```{r}
#first combine the COVID patient plots
plot_patient <- plot_grid(plot2, plot1, nrow = 2, rel_heights = c(1/3, 2/3), align= "v", axis="rlbt")

dir <- "scaled_diff_select_plots/"
plot_name <- paste(dir, "FP_region_patients_combined.jpg", sep = "")
ggsave(plot_name, plot_patient, width = 20, height = 12)
```


linker loop/HR2 peptide region, selected patient samples, scaled diff sel
```{r}

dir <- "scaled_diff_select_plots/"

ID.labs <- c("30 days p.s.o.", "60 days p.s.o.")
names(ID.labs) <- c("conv outpatient 30d", "conv outpatient 60d")

 plot1 <- master %>% 
    filter(Protein != "S1D614G") %>%
    filter(participant_ID %in% c("5", "9","13", "18")) %>%
    filter(Correct_Loc %in% 1135:1195) %>%
   ggplot( aes(x=Correct_Loc, y=aa_sub, fill=scaled_diff_sel, width=.90, height=.90)) + 
   geom_tile() +
  geom_point(aes(size=is_wt), stroke = 0) +
  scale_size_manual(values=c(0, 2)) +
  scale_fill_gradientn(colours = c("red","white","darkblue"), values = rescale(c(-200,0,100)), guide = "colorbar", limits=c(-200,100), na.value ="lightgrey") + 
  labs(x="SARS-CoV-2 S Protein Position", y="AA Substitution", fill = "Scaled differential \nselection value\n") +
  theme_bw(base_size=16) +
  facet_grid(patient_status ~ participant_ID, labeller = labeller(patient_status = ID.labs)) +
  rotate_x_text(angle=45) +
  theme(plot.margin = unit(c(-0.5, 0, 0, 0), "cm"), strip.background = element_blank(), strip.text.x = element_blank(), strip.text.y = element_text(face = "bold", size = 16), legend.title.align=0.5, panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
   guides(size = FALSE) + 
   scale_x_continuous(breaks = seq(1135, 1195, by = 5), expand = c(0.02,0.02)) 

 plot_name <- paste(dir, "HR2_region_patients_scaleddiffsel.jpg", sep = "")
  
  ggsave(plot_name, plot1, width = 20, height = 8)
  
```

Now make an enrichment plot for the selected patient samples within the HR2 region
```{r}
plot2 <- master %>% 
  filter(Protein != "S1D614G") %>% 
  filter(participant_ID %in% c("5", "9","13", "18")) %>%
  filter(is_wt == TRUE) %>%
  filter(Correct_Loc %in% 1135:1195) %>%
  ggplot( aes(x=Correct_Loc, y=enrichment, color=participant_ID, linetype=sample_day)) + 
    geom_line(alpha=0.8) + 
  theme_bw(base_size = 16) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  facet_wrap(~participant_ID, nrow=1) +
  ylab("Enrichment of WT peptides") +
  labs(linetype = "Sample timepoint") +
  ylim(-8,80) +
  scale_color_manual(values = Color_ID)  +
  guides(color = FALSE) +
  theme(plot.margin = unit(c(0, 0, 0, 0), "cm"), strip.background = element_blank(), strip.text.x = element_text(face="bold", size = 20), legend.title.align=0.5, panel.border = element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.title.x=element_blank()) +
   scale_linetype_discrete(labels=c("30 days p.s.o.", "60 days p.s.o."))+ 
   scale_x_continuous(breaks = seq(1135, 1195, by = 5), expand = c(0.02,0.02))+
   coord_cartesian(clip="off") +
   annotate("rect", xmin=1135, xmax=1162.5, ymin=-8, ymax=-2, alpha=0.4, fill="grey60")+
  annotate("text", x=1148.5, y=-5, label="LINKER REGION", fontface="bold")+
    annotate("rect", xmin=1162.6, xmax=1195, ymin=-8, ymax=-2, alpha=0.4, fill="grey25")+
  annotate("text", x=1179, y=-5, label="HR2", fontface="bold")


dir <- "scaled_diff_select_plots/"

plot_name <- paste(dir, "HR2_region_patients_enrichment.jpg", sep = "")

ggsave(plot_name, plot2, width = 20, height = 5)

```

Now, combine the two plots into one
```{r}
#first combine the COVID patient plots
plot_patient <- plot_grid(plot2, plot1, nrow = 2, rel_heights = c(1/3, 2/3), align= "v", axis="rlbt")

dir <- "scaled_diff_select_plots/"
plot_name <- paste(dir, "HR2_region_patients_combined.jpg", sep = "")
ggsave(plot_name, plot_patient, width = 20, height = 12)
```

Other regions, selected patient samples, scaled diff sel

First make the enrichment plot and heatmap for the NTD region
```{r}

dir <- "scaled_diff_select_plots/"

ID.labs <- c("30 days p.s.o.", "60 days p.s.o.")
names(ID.labs) <- c("conv outpatient 30d", "conv outpatient 60d")

#First plot heatmap
 plot_NTD_heatmap <- master %>% 
    filter(Protein != "S1D614G") %>%
    filter(participant_ID  == "12") %>%
    filter(Correct_Loc %in% 245:285) %>%
    ggplot( aes(x=Correct_Loc, y=aa_sub, fill=scaled_diff_sel, width=.90, height=.90)) + 
      geom_tile() +
  geom_point(aes(size=is_wt), stroke = 0) +
  scale_size_manual(values=c(0, 2)) +
  scale_fill_gradientn(colours = c("red","white","darkblue"), values = rescale(c(-250,0,100)), guide = "colorbar", limits=c(-250,100), na.value ="lightgrey") + 
  labs(x="SARS-CoV-2 S Protein Position", y="AA Substitution", fill = "Scaled differential \nselection value\n") +
  theme_bw(base_size=16) +
  rotate_x_text(angle=45) +
  facet_grid(patient_status ~ participant_ID, labeller = labeller(patient_status = ID.labs)) +
  theme(legend.position = "none", axis.title.x=element_blank(), plot.margin = unit(c(-0.4, 0, 0, 0), "cm"), strip.background = element_blank(), strip.text.x = element_blank(), strip.text.y = element_blank(), legend.title.align=0.5, panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
   guides(size = FALSE) + 
   scale_x_continuous(breaks = seq(245, 285, by = 5), expand = c(0.02,0.02)) 
  
 plot_name <- paste(dir, "NTD_region_patients_scaleddiffsel.jpg", sep = "")
  
  ggsave(plot_name, plot_NTD_heatmap, width = 5, height = 8)
  
  
#Then plot enrichment
  plot_NTD_enrichment <- master %>% 
  filter(Protein != "S1D614G") %>% 
  filter(participant_ID  == "12") %>%
  filter(is_wt == TRUE) %>%
  filter(Correct_Loc %in% 245:285) %>%
  ggplot(aes(x=Correct_Loc, y=enrichment, color=participant_ID, linetype=patient_status)) +
    geom_line(alpha=0.8) + 
  theme_bw(base_size = 16) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  facet_wrap(~participant_ID, nrow=1) +
  ylab("Enrichment of WT peptides") +
  labs(linetype = "Sample timepoint") +
  ylim(-8,80) +
  scale_color_manual(values = Color_ID)  +
  guides(color = FALSE) +
  theme(legend.position = "none", plot.margin = unit(c(0, 0, 0, 0), "cm"), strip.background = element_blank(), strip.text.x = element_text(face="bold", size = 20), legend.title.align=0.5, panel.border = element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.title.x=element_blank()) +
   scale_linetype_discrete(labels=c("30 days p.s.o.", "60 days p.s.o."))+ 
   scale_x_continuous(breaks = seq(245, 285, by = 5), expand = c(0.02,0.02))+
   coord_cartesian(clip="off") +
   annotate("rect", xmin=245, xmax=285, ymin=-8, ymax=-2, alpha=0.4, color="grey70", fill="grey70")+
  annotate("text", x=265, y=-5, label="NTD", fontface="bold")



dir <- "scaled_diff_select_plots/"

plot_name <- paste(dir, "NTD_region_patients_enrichment.jpg", sep = "")

ggsave(plot_name, plot_NTD_enrichment, width = 5, height = 5)


#Then combine the NTD plots
NTD_plot <- plot_grid(plot_NTD_enrichment, plot_NTD_heatmap, ncol = 1, rel_heights = c(1/3, 2/3), align= "v", axis="rlbt") 

print(NTD_plot)
```


Now make the enrichment plot and heatmap for the RBD region

```{r}

dir <- "scaled_diff_select_plots/"

ID.labs <- c("30 days p.s.o.", "60 days p.s.o.")
names(ID.labs) <- c("conv outpatient 30d", "conv outpatient 60d")

#First plot heatmap
 plot_RBD_heatmap <- master %>% 
    filter(Protein != "S1D614G") %>%
    filter(participant_ID  == "15") %>%
    filter(Correct_Loc %in% 470:510) %>%
    ggplot( aes(x=Correct_Loc, y=aa_sub, fill=scaled_diff_sel, width=.90, height=.90)) + 
   geom_tile() +
  geom_point(aes(size=is_wt), stroke = 0) +
  scale_size_manual(values=c(0, 2)) +
  scale_fill_gradientn(colours = c("red","white","darkblue"), values = rescale(c(-250,0,100)), guide = "colorbar", limits=c(-250,100), na.value ="lightgrey") + 
  labs(x="SARS-CoV-2 S Protein Position", y="AA Substitution", fill = "Scaled differential \nselection value\n") +
  theme_bw(base_size=16) +
  rotate_x_text(angle=45) +
  facet_grid(patient_status ~ participant_ID, labeller = labeller(patient_status = ID.labs)) +
  theme(legend.position = "none", axis.text.y = element_blank(),
  axis.ticks.y = element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), plot.margin = unit(c(-0.4, 0, 0, 0), "cm"), strip.background = element_blank(), strip.text.x = element_blank(), strip.text.y = element_blank(), legend.title.align=0.5, panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
   guides(size = FALSE) + 
   scale_x_continuous(breaks = seq(470, 510, by = 5), expand = c(0.02,0.02)) 


 plot_name <- paste(dir, "RBD_region_patients_scaleddiffsel.jpg", sep = "")
  
  ggsave(plot_name, plot_RBD_heatmap, width = 5, height = 8)
  
  
#Then plot enrichment
  plot_RBD_enrichment <- master %>% 
  filter(Protein != "S1D614G") %>% 
  filter(participant_ID  == "15") %>%
  filter(is_wt == TRUE) %>%
  filter(Correct_Loc %in% 470:510) %>%
  ggplot( aes(x=Correct_Loc, y=enrichment, color=participant_ID, linetype=patient_status)) + 
  geom_line(alpha=0.8) + 
  theme_bw(base_size = 16) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  facet_wrap(~participant_ID, nrow=1) +
  ylab("Enrichment of WT peptides") +
  labs(linetype = "Sample timepoint") +
  ylim(-8,80) +
  scale_color_manual(values = Color_ID)  +
  guides(color = FALSE) +
  theme(legend.position = "none", axis.text.y = element_blank(),
  axis.ticks.y = element_blank(), plot.margin = unit(c(0, 0, 0, 0), "cm"), strip.background = element_blank(), strip.text.x = element_text(face="bold", size = 20), legend.title.align=0.5, panel.border = element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank()) +
   scale_linetype_discrete(labels=c("30 days p.s.o.", "60 days p.s.o."))+ 
   scale_x_continuous(breaks = seq(470, 510, by = 5), expand = c(0.02,0.02))+
   coord_cartesian(clip="off") +
   annotate("rect", xmin=470, xmax=510, ymin=-8, ymax=-2, alpha=0.4, color="grey50", fill="grey50")+
  annotate("text", x=490, y=-5, label="RBD", fontface="bold")


dir <- "scaled_diff_select_plots/"

plot_name <- paste(dir, "RBD_region_patients_enrichment.jpg", sep = "")

ggsave(plot_name, plot_RBD_enrichment, width = 5, height = 5)


#Then combine the RBD plots
RBD_plot <- plot_grid(plot_RBD_enrichment, plot_RBD_heatmap, ncol = 1, rel_heights = c(1/3, 2/3), align= "v", axis="rlbt") 


```


Now make the enrichment plot and heatmap for the CTD1 region

```{r}

dir <- "scaled_diff_select_plots/"

ID.labs <- c("30 days p.s.o.", "60 days p.s.o.")
names(ID.labs) <- c("conv outpatient 30d", "conv outpatient 60d")

#First plot heatmap
 plot_CTD1_heatmap <- master %>% 
    filter(Protein != "S1D614G") %>%
    filter(participant_ID  == "15") %>%
    filter(Correct_Loc %in% 540:580) %>%
    ggplot( aes(x=Correct_Loc, y=aa_sub, fill=scaled_diff_sel, width=.90, height=.90)) + 
    geom_tile() +
  geom_point(aes(size=is_wt), stroke = 0) +
  scale_size_manual(values=c(0, 2)) +
  scale_fill_gradientn(colours = c("red","white","darkblue"), values = rescale(c(-250,0,100)), guide = "colorbar", limits=c(-250,100), na.value ="lightgrey") + 
  labs(x="SARS-CoV-2 S Protein Position", y="AA Substitution", fill = "Scaled differential \nselection value\n") +
  theme_bw(base_size=16) +
  rotate_x_text(angle=45) +
  facet_grid(patient_status ~ participant_ID, labeller = labeller(patient_status = ID.labs)) +
  theme(legend.position = "none", axis.text.y = element_blank(),
  axis.ticks.y = element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), plot.margin = unit(c(-0.4, 0, 0, 0), "cm"), strip.background = element_blank(), strip.text.x = element_blank(), strip.text.y = element_blank(), legend.title.align=0.5, panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
   guides(size = FALSE) + 
   scale_x_continuous(breaks = seq(540, 580, by = 5), expand = c(0.02,0.02)) 


 plot_name <- paste(dir, "CTD1_region_patients_scaleddiffsel.jpg", sep = "")
  
  ggsave(plot_name, plot_CTD1_heatmap, width = 5, height = 8)
  
  
#Then plot enrichment
  plot_CTD1_enrichment <- master %>% 
  filter(Protein != "S1D614G") %>% 
  filter(participant_ID  == "15") %>%
  filter(is_wt == TRUE) %>%
  filter(Correct_Loc %in% 540:580) %>%
  ggplot( aes(x=Correct_Loc, y=enrichment, color=participant_ID, linetype=patient_status)) + 
  geom_line(alpha=0.8) + 
  theme_bw(base_size = 16) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  facet_wrap(~participant_ID, nrow=1) +
  ylab("Enrichment of WT peptides") +
  labs(linetype = "Sample timepoint") +
  ylim(-8,80) +
  scale_color_manual(values = Color_ID)  +
  guides(color = FALSE) +
  theme(legend.position = "none", axis.text.y = element_blank(),
  axis.ticks.y = element_blank(), plot.margin = unit(c(0, 0, 0, 0), "cm"), strip.background = element_blank(), strip.text.x = element_text(face="bold", size = 20), legend.title.align=0.5, panel.border = element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank()) +
   scale_linetype_discrete(labels=c("30 days p.s.o.", "60 days p.s.o."))+ 
   scale_x_continuous(breaks = seq(540, 580, by = 5), expand = c(0.02,0.02))+
   coord_cartesian(clip="off") +
   annotate("rect", xmin=540, xmax=580, ymin=-8, ymax=-2, alpha=0.4, color="grey30", fill="grey30")+
  annotate("text", x=560, y=-5, label="CTD", fontface="bold")


dir <- "scaled_diff_select_plots/"

plot_name <- paste(dir, "CTD1_region_patients_enrichment.jpg", sep = "")

ggsave(plot_name, plot_CTD1_enrichment, width = 5, height = 5)


#Then combine the CTD1 plots
CTD1_plot <- plot_grid(plot_CTD1_enrichment, plot_CTD1_heatmap, ncol = 1, rel_heights = c(1/3, 2/3), align= "v", axis="rlbt") 


```

Now make the enrichment plot and heatmap for the CTD2 region

```{r}

dir <- "scaled_diff_select_plots/"

ID.labs <- c("30 days p.s.o.", "60 days p.s.o.")
names(ID.labs) <- c("conv outpatient 30d", "conv outpatient 60d")

#First plot heatmap, with legend
 plot_CTD2_heatmap <- master %>% 
    filter(Protein != "S1D614G") %>%
    filter(participant_ID  == "3") %>%
    filter(Correct_Loc %in% 615:650) %>%
    ggplot( aes(x=Correct_Loc, y=aa_sub, fill=scaled_diff_sel, width=.90, height=.90)) + 
  geom_tile() +
  geom_point(aes(size=is_wt), stroke = 0) +
  scale_size_manual(values=c(0, 2)) +
  scale_fill_gradientn(colours = c("red","white","darkblue"), values = rescale(c(-250,0,100)), guide = "colorbar", limits=c(-250,100), na.value ="lightgrey") + 
  labs(x="SARS-CoV-2 S Protein Position", y="AA Substitution", fill = "Scaled differential \nselection value\n") +
  theme_bw(base_size=16) +
  rotate_x_text(angle=45) +
  facet_grid(patient_status ~ participant_ID, labeller = labeller(patient_status = ID.labs)) +
  theme(axis.title.x=element_blank(), axis.text.y = element_blank(),
  axis.ticks.y = element_blank(), axis.title.y=element_blank(), plot.margin = unit(c(-0.4, 0, 0, 0), "cm"), strip.background = element_blank(), strip.text.x = element_blank(), strip.text.y = element_text(face = "bold", size = 16), legend.title.align=0.5, panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
   guides(size = FALSE) + 
   scale_x_continuous(breaks = seq(615, 650, by = 5), expand = c(0.02,0.02)) 

 plot_name <- paste(dir, "CTD2_region_patients_scaleddiffsel.jpg", sep = "")
  
  ggsave(plot_name, plot_CTD2_heatmap, width = 5, height = 8)
  
  
#Then plot enrichment
  plot_CTD2_enrichment <- master %>% 
  filter(Protein != "S1D614G") %>% 
  filter(participant_ID  == "3") %>%
  filter(is_wt == TRUE) %>%
  filter(Correct_Loc %in% 615:650) %>%
  ggplot( aes(x=Correct_Loc, y=enrichment, color=participant_ID, linetype=patient_status)) + 
  geom_line(alpha=0.8) + 
  theme_bw(base_size = 16) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  facet_wrap(~participant_ID, nrow=1) +
  ylab("Enrichment of WT peptides") +
  labs(linetype = "Sample timepoint") +
  ylim(-8,80) +
  scale_color_manual(values = Color_ID)  +
  guides(color = FALSE) +
  theme(plot.margin = unit(c(0, 0, 0, 0), "cm"), axis.text.y = element_blank(),
  axis.ticks.y = element_blank(), strip.background = element_blank(), strip.text.x = element_text(face="bold", size = 20), legend.title.align=0.5, panel.border = element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank()) +
   scale_linetype_discrete(labels=c("30 days p.s.o.", "60 days p.s.o."))+ 
   scale_x_continuous(breaks = seq(615, 650, by = 5), expand = c(0.02,0.02))+
   coord_cartesian(clip="off") +
   annotate("rect", xmin=615, xmax=650, ymin=-8, ymax=-2, alpha=0.4, color="grey30", fill="grey30")+
  annotate("text", x=632.5, y=-5, label="CTD", fontface="bold")


dir <- "scaled_diff_select_plots/"

plot_name <- paste(dir, "CTD2_region_patients_enrichment.jpg", sep = "")

ggsave(plot_name, plot_CTD2_enrichment, width = 5, height = 5)


#Then combine the CTD2 plots
CTD2_plot <- plot_grid(plot_CTD2_enrichment, plot_CTD2_heatmap, ncol = 1, rel_heights = c(1/3, 2/3), align= "v", axis="rlbt") 


```

Now, combine all enrichment and heatmaps from the "other" regions, add a legend
```{r}
#first combine the COVID patient plots
plot <- plot_grid(NTD_plot, RBD_plot, CTD1_plot, CTD2_plot, nrow = 1, rel_widths = c(1.1, 1, 1, 1.58))

#Annotate the plot with x axis and y axis labels
plot <- annotate_figure(plot, bottom = text_grob("SARS-CoV-2 S Protein Position", size = 14))

dir <- "scaled_diff_select_plots/"

plot_name <- paste(dir, "other_regions_patients_enrichment.jpg", sep = "")

ggsave(plot_name, plot, width = 20, height = 12)

```

## Compare findings to circulating variants based on GSIAID sequencing data


**Plot the enrichment of all mutant peptides that span the D614G site**
```{r}
dir.create("GSIAID_plots")

#The peptides are 31 aa long, so aa 599-629 are those sites. Need to also remove site D614 since those will have the same value between the two backgrounds
D614G_data <- master %>% 
  filter(Correct_Loc %in% 599:619) %>%
  filter(Correct_Loc != 614) %>% 
  filter(patient_status == "conv outpatient 60d") %>%
  filter(is_wt == FALSE) %>%
  select(enrichment, Correct_Loc, participant_ID, Protein, aa_sub) %>%
  pivot_wider(names_from = "Protein", values_from = "enrichment") %>%
  filter(!is.na(S1) & !is.na(S1D614G))

plot <- D614G_data %>%
  ggpaired(cond1="S1", cond2="S1D614G", color="participant_ID", line.color = "gray", line.size = 0.4, width=0) +
  facet_wrap(~participant_ID, nrow=2) +
  stat_compare_means(paired=TRUE, label.y= 40, label.x =1.46, aes(label = ..p.signif..)) +
  theme_classic(base_size = 18) +
  theme(panel.border = element_rect(colour = "black", fill=NA, size=1),  axis.text=element_text(size=18,face="bold"), legend.position = "none") +
  scale_color_manual(values = Color_ID)  +
  ylim(-1,42) +
  xlab("") +
  ylab("Enrichment") +
  theme(strip.background = element_blank(), strip.text.x = element_text(face="bold", size = 18), panel.border = element_rect(colour = "black", fill=NA, size=1)) +
  scale_x_discrete(labels= c("D614", "G614"), expand = c(0.2,0.2))


dir <- "GSIAID_plots/"
  
plot_name <- paste(dir, "D614G_comparison_enrichment_60d.jpg", sep = "")

ggsave(plot_name, plot, width = 18, height = 8)


print(plot)
```



Test the effect size of D614G mutants
```{r}
D614G_data <- master %>% 
  filter(Correct_Loc %in% 599:619) %>%
  filter(Correct_Loc != 614) %>% 
  filter(patient_status == "conv outpatient 60d") %>%
  filter(is_wt == FALSE) 

D614G_data %>%
  group_by(participant_ID) %>%
  wilcox_effsize(enrichment ~ Protein, paired=TRUE)
```
**plot the avg site scaled diff sel vs the site mutational entropy**

First, import the tsv file from LANL [here](https://cov.lanl.gov/components/sequence/COV/int_sites_tbls.comp?t=2&l=). This was downloaded on 11/9/20
```{r}
GISAID_table <- read_tsv("per_site_variation.tsv") %>% 
  select(Correct_Loc = "Ref seq position", site_mut_entropy= Entropy, site='Ref seq') %>% 
  filter(site != '-') %>%
  select(-site)

```

Then get the averaged scaled diff sel values, by site
```{r}
master_averaged <- master %>% filter(Protein != "S1D614G") %>%
  filter(is_wt == FALSE) %>%
  filter(sample_day == "60d") %>%
  group_by(Correct_Loc, participant_ID) %>%
  summarize(avg_scaled_diff_sel = mean(scaled_diff_sel, na.rm=T)) %>%
  ungroup()
```

Now combine the tables
```{r}

master_averaged <- left_join(master_averaged, GISAID_table)

#Now, need a column labeling the sites by domain
master_averaged <- master_averaged %>%
  mutate(epitope = case_when(
    Correct_Loc %in% 807:837 ~ "FP epitope", 
    Correct_Loc %in% 1135:1162 ~ "linker region epitope",
    Correct_Loc %in% 1163:1193 ~ "HR2 epitope", 
    TRUE ~ "other"
  ))

#Now refactor the epitope labels
master_averaged$epitope <- factor(master_averaged$epitope, levels = c("FP epitope", "linker region epitope", "HR2 epitope", "other"))
```


Now, plot the summed scaled diff sel by the site mutational entropy, faceting by patient
```{r}

GISAID_scatterplot <- master_averaged %>%
  filter(!is.na(avg_scaled_diff_sel)) %>%
  ggplot(aes(x=site_mut_entropy, y=avg_scaled_diff_sel, color=epitope)) +
  geom_point(alpha=0.5, size=3) +
  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
              labels = trans_format("log10", math_format(10^.x))) + 
  facet_wrap(~participant_ID,  nrow=3, scales="free_y") +
  theme_classic(base_size = 20) +
  theme(strip.background = element_blank(), strip.text.x = element_text(face="bold", size = 18), panel.border = element_rect(colour = "black", fill=NA, size=1), legend.position = "bottom") +
  xlab("Site mutational entropy") +
  ylab("Scaled differential selection, averaged by site") +
  scale_color_manual(name="Location of site", breaks=c("FP epitope", "linker region epitope", "HR2 epitope", "other"), values=c( "green", "orange", "red", "black")) +
  geom_vline(xintercept = 0.02, linetype="dotted", 
                color = "grey", alpha=0.7, size=1.5) +
  geom_text_repel(data=subset(master_averaged, avg_scaled_diff_sel < -20 & site_mut_entropy >= 0.02), nudge_x = 0.01, nudge_y = -2,  aes(label = Correct_Loc, color=epitope)) 


dir <- "GSIAID_plots/"
  
plot_name <- paste(dir, "mut_entropy_all_sites_sumscaleddiffsel.jpg", sep = "")

ggsave(plot_name, GISAID_scatterplot, width = 18, height = 10)

print(GISAID_scatterplot)
```
Now, average the differential selection values by site, across all patients, and plot aggregated data
```{r}
diffsel_averaged <- master_averaged %>% 
  group_by(Correct_Loc) %>%
  summarize(avg_all_scaled_diff_sel = mean(avg_scaled_diff_sel, na.rm=T)) %>%
  ungroup()
```

Add back in site mut entropy values, and epitope labels
```{r}
diffsel_averaged <- left_join(diffsel_averaged, GISAID_table) %>%
  mutate(epitope = case_when(
    Correct_Loc %in% 807:837 ~ "FP epitope", 
    Correct_Loc %in% 1135:1162 ~ "linker region epitope",
    Correct_Loc %in% 1163:1193 ~ "HR2 epitope", 
    TRUE ~ "other"
  ))

#Now refactor the epitope labels
diffsel_averaged$epitope <- factor(diffsel_averaged$epitope, levels = c("FP epitope", "linker region epitope", "HR2 epitope", "other"))
```



Now plot comparison between differential selection and mutational entropy, averaged across all patients
```{r}
GISAID_scatterplot_avg <- diffsel_averaged %>%
  ggplot(aes(x=site_mut_entropy, y=avg_all_scaled_diff_sel, color=epitope)) +
  geom_point(alpha=0.5, size=3) +
   scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
              labels = trans_format("log10", math_format(10^.x))) + 
  theme_classic(base_size = 20) +
  theme(strip.background = element_blank(), strip.text.x = element_text(face="bold", size = 18), panel.border = element_rect(colour = "black", fill=NA, size=1), legend.position = "bottom") +
  xlab("Site mutational entropy") +
  ylab("Scaled differential selection, averaged by site") +
  scale_color_manual(name="Location of site", breaks=c("FP epitope", "linker region epitope", "HR2 epitope", "other"), values=c( "green", "orange", "red", "black")) +
  geom_vline(xintercept = 0.02, linetype="dotted", 
                color = "grey", alpha=0.5, size=1.5) +
  geom_text_repel(data=subset(diffsel_averaged, avg_all_scaled_diff_sel < -20 & site_mut_entropy >= 0.02), nudge_x = 0.01, nudge_y = -2,  aes(label = Correct_Loc, color=epitope)) 


dir <- "GSIAID_plots/"
  
plot_name <- paste(dir, "mut_entropy_all_sites_sumscaleddiffsel_avg.jpg", sep = "")

ggsave(plot_name, GISAID_scatterplot_avg, width = 10, height = 8)

print(GISAID_scatterplot_avg)

```



**Plot of all commonly circulating Spike mutants and their scaled diff sel values**

Will take the scaled diff sel values of the the mutants and compare against example mutants
First need to get data from LANL on the most common mutants
```{r}
GISIAD_data <- read_csv("sites_common_mutations.csv")

GISIAD_data <- GISIAD_data %>% 
  group_by(pos) %>%
  top_n(1, global_frequency) %>%
  filter(mut != "-") %>%
  ungroup() %>%
  #select the top 20 most common mutants
  top_n(20, global_frequency) %>%
  mutate(label = paste(ori, pos, mut, sep="")) %>%
  mutate(GISAIDmutants = "variant") %>%
  select(aa_sub = mut, Correct_Loc = pos, GISAIDmutants, label) %>%
  mutate(label = factor(label, label))
  
```
Now make new table with diff sel data
```{r}
GISIAD_data <- left_join(master, GISIAD_data) %>% 
  filter(!is.na(GISAIDmutants )) %>% 
  filter(Protein != "S1D614G") %>% 
  filter(patient_status == "conv outpatient 60d")
```

Now plot the scaled diff sel values for commonly found variants
```{r}
plot1 <- GISIAD_data %>% 
  ggplot(aes(x = reorder(label, desc(label)), y=scaled_diff_sel, color = participant_ID)) +
  geom_jitter(size=3, width=0.1) +
  theme_classic(base_size = 16) +
  ggtitle("Commonly circulating mutants \nidentified by GISAID") + 
  ylab("Scaled differential selection value") +
  labs(color = "Patient ID") +
  scale_color_manual(values = Color_ID) +
  theme(panel.border = element_rect(colour = "black", fill=NA, size=1),legend.position = "none", axis.title.x=element_blank(), plot.title = element_text(hjust = 0.5),  axis.title.y=element_blank(), axis.text=element_text(size=14)) +
  ylim(-230, 5) +
  scale_x_discrete(position = "top") +
  coord_flip() 
  
print(plot1)

dir <- "GSIAID_plots/"
  
plot_name <- paste(dir, "mutation_level_comparison_GISAIDmutants_scaleddiffsel.jpg", sep = "")

ggsave(plot_name, plot1, width = 6, height = 7)
```


Now plot selected mutants found by Phage-DMS
```{r}
dir.create("GSIAID_plots")

GISIAD_data <- master %>%
  mutate(ctrl_mutants = case_when(
    Correct_Loc == "819" & aa_sub == "K" ~ "variant",
    Correct_Loc == "1156" & aa_sub == "G" ~ "variant")) %>% 
  filter(!is.na(ctrl_mutants )) %>% 
  filter(Protein != "S1D614G") %>% 
  filter(patient_status == "conv outpatient 60d") %>%
  mutate(label = case_when(
    Correct_Loc == "819" ~ "E819K", 
    Correct_Loc == "1156" ~ "F1156G"))

plot2 <- GISIAD_data %>% 
  ggplot(aes(x = reorder(label, desc(label)), y=scaled_diff_sel, color = participant_ID)) +
  geom_jitter(size=3, width=0.1) +
  theme_classic(base_size = 16) +
  ggtitle("Example escape mutants \nidentified by Phage-DMS") + 
  coord_flip() +
  ylab("Scaled differential selection value") +
  labs(color = "Patient ID") +
  scale_color_manual(values = Color_ID) +
  theme(panel.border = element_rect(colour = "black", fill=NA, size=1), axis.title.y=element_blank(), plot.title = element_text(hjust = 0.5), legend.position = "none", axis.text=element_text(size=14))+
  ylim(-230, 5) +
  scale_x_discrete(position = "top") 
  
print(plot2)

dir <- "GSIAID_plots/"
  
plot_name <- paste(dir, "mutation_level_comparison_examplemutants_scaleddiffsel.jpg", sep = "")

ggsave(plot_name, plot2, width = 5, height = 2)

```


```{r}
#Then combine the  plots
GISAID_plot <- plot_grid(plot1, plot2, ncol = 1, rel_heights =c(.78, .22), align= "v", axis="rlbt") 

#Annotate the plot with legend
legend <- get_legend(plot2 + theme(legend.position="right"))

#Add legend to plot
plot <- plot_grid(legend, GISAID_plot, rel_widths = c(1, 4))


dir <- "GSIAID_plots/"
  
plot_name <- paste(dir, "mutation_level_comparison_scaleddiffsel.jpg", sep = "")

ggsave(plot_name, plot, width = 8, height = 11)
```


## Finally, need to export data for viewing in DMS-view
```{r}
dir.create("dms-view")

#First create a column with the wildtype aa
dms_view_data <- master %>%
  #I'm filtering the data to only show the results from the WT Wuhan strain
  filter(Protein != "S1D614G") %>%
  mutate(wildtype = case_when(is_wt == TRUE ~ aa_sub)) %>%
  group_by(Correct_Loc) %>%
  fill(wildtype, .direction="downup")

#Then rename some columns in order to match dms-view requirements
dms_view_data <- dms_view_data %>%
  select(site = Correct_Loc, wildtype, mutation = aa_sub, condition = participant_timepoint, site_WT_enrichment = enrichment, mut_scaled_diff_sel = scaled_diff_sel) %>%
  add_column(protein_chain = "A B C") %>%
  mutate(label_site = site) %>%
  mutate(protein_site = site) %>%
  filter(!is.na(mut_scaled_diff_sel))

#Now export this to a csv
write.csv(dms_view_data,file = "dms-view/Spike_DMS_view_data.csv")
```

##Also, export WT binding data to correlate with neutralization data
First need to extract the max binding enrichment of WT peptides in the FP and HR2 regions, then export to folder with neutralization data
```{r}
binding_data <- master %>% 
  filter(is_wt == TRUE) %>%
  mutate(epitope = case_when(
    Correct_Loc %in% 807:837 ~ "FP region", 
    Correct_Loc %in% 1135:1193 ~ "HR2 region",
    TRUE ~ "other")) %>%
  filter(epitope != "other") %>%
  filter(sample_day == "30d") %>%
  group_by(participant_ID, epitope) %>%
  top_n(1, enrichment) %>%
  select(participant_ID, epitope, enrichment)

#Now export this to a csv
write.csv(binding_data,file = "neutralization_data/Spike_DMS_binding_data.csv")

```

##Now, import AUC data so that I can plot it.
```{r}
AUC_data <- read_csv("ELISA_AUC_data.csv")

AUC_data$participant_ID <- as.factor(AUC_data$participant_ID)

AUC_data <- AUC_data %>% mutate(participant_ID = fct_relevel(participant_ID, "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18"))
```

Now plot paired plot of RBD ELISA AUC
```{r}
dir.create("ELISA_plots")

plot_RBD <- AUC_data %>%
  pivot_wider(names_from = condition, values_from= AUC) %>%
  filter(antigen == "RBD") %>%
  ggpaired(cond1="mock", cond2="depleted", color="participant_ID", line.color = "gray", line.size = 0.4, width=0, point.size = 3) +
  theme_classic(base_size = 20) +
  scale_color_manual(values = Color_ID)  +
  ylab("Binding activity (ELISA AUC)") +
  ggtitle("RBD") +
  theme(plot.title = element_text(hjust = 0.5), legend.position = "none", axis.title.x=element_blank()) +
  geom_hline(yintercept=0.059, linetype="dashed") 

print(plot_RBD)

```
Now make plot with just Spike
```{r}
plot_Spike <- AUC_data %>%
  pivot_wider(names_from = condition, values_from= AUC) %>%
  filter(antigen == "Spike") %>%
  ggpaired(cond1="mock", cond2="depleted", color="participant_ID", line.color = "gray", line.size = 0.4, width=0, point.size = 3) +
  theme_classic(base_size = 20) +
  scale_color_manual(values = Color_ID)  +
  ylab("Binding activity (ELISA AUC)") +
  ggtitle("Spike") +
  theme(plot.title = element_text(hjust = 0.5), legend.position = "none", axis.title.x=element_blank(),  axis.title.y=element_blank())+
  geom_hline(yintercept=1.08, linetype="dashed")  

print(plot_Spike)

```
Now make plot with just S2
```{r}
plot_S2 <- AUC_data %>%
  filter(antigen == "S2") %>%
  ggplot(aes(x=condition, y=AUC, color=participant_ID)) +
  geom_jitter(size = 3, width = 0.2) +
  theme_classic(base_size = 20) +
  scale_color_manual(values = Color_ID)  +
  ylab("Binding activity (ELISA AUC)") +
  ggtitle("S2") +
  theme(plot.title = element_text(hjust = 0.5), axis.title.x=element_blank(),  axis.title.y=element_blank())+
  geom_hline(yintercept=0.058, linetype="dashed")  +
  labs(color = "Patient ID") +
  guides(col = guide_legend(ncol = 2))

print(plot_S2)



```

```{r}
#first combine the COVID patient plots
plot_AUC <- plot_grid(plot_RBD, plot_Spike, plot_S2, nrow = 1)

dir <- "ELISA_plots/"
plot_name <- paste(dir, "ELISA_AUC_plots.jpg", sep = "")
ggsave(plot_name, plot_AUC, width = 14, height = 6)
```

